index.bs

```diff
@@ -931,6 +933,50 @@ authorizing an authenticator with which to complete the operation.
 </div>
 
 
+### Advertise Platform Authenticator Support - isPlatformAuthenticatorReady() method ### {#isPlatformAuthenticatorReady}
+
+<div link-for-hint="WebAuthentication/isPlatformAuthenticatorReady">
+This method can be used by [=[RPS]=] to inform the user that they can create a new credential with one or more platform authenticators.
+The method gives [=[RPS]=] the ability to inform the client that they support registering credentials with [=platform authenticators=]
+while giving users control over whether this information is appropriate to provide to the RP.
+When the method is invoked, the [=client=] will search for the existence of a [=platform authenticator=]
+capable of registering credentials.
+Upon finding one, the [=client=] will determine whether the user would be interested in
+creating a credential based on a number of factors, which may include the user's explicitly stated intentions.
+If such interest is found, the promise will be resolved with the value of <i>True</i>.
+Based on the result, the [=[RP]=] can take further actions to guide the user to create a credential.
+
+This method has no arguments and returns a boolean value.
+
+<pre class="idl">
+    [SecureContext]
+    partial interface PublicKeyCredential {
+        [Unscopable] Promise < boolean > isPlatformAuthenticatorReady();
```

Comment:
@selfissued @mikewest I thought about this question a few times in the past. The one question to be answered here is whether sharing the PlatformAuthenticatorReady information counts as fingerprinting and how much we want to tolerate the fingerprinting concern. 

Typically, for any personally identifiable information to be shared on the web, we would want to make sure we get user consent, such as prompting permission for geolocation etc. In case, the information is even more sensitive than geo. Unlike geo where the device almost always functions when it is hooked up to the client, most biometric authenticators require the user to enter their biometric info for a few times before the authenticators become ready. For example, Windows would ask the user to create a PIN first and then swipe their fingers multiple times. If the function resolves without UX or becomes an attribute that can be silently retrieved by any script, any site on the world would know (or at least have a good guess) the user has entered some sort of biometric info on the client. Of course we then enter into the discussion of whether the user actually cares we share this info. 

If the method becomes an attribute, the method would not prompt UI when being requested for this information and can become a privacy concern. 

It'd probably make sense to define an attribute on top of a method. In this sense, it means that after the user shows the interest, the platform will remember the choice. When the page loads in the future, the developer can check the attribute and can decide what UI to show. 

A note on the syntax proposed above, I am not sure if there needs to be a promise in there when the info is on the device. But I guess async is always good on the web.